2020/day3.clj

```diff
@@ -1,89 +0,0 @@
-(ns advent-of-code.2020.day3
-  (:require [clojure.string :as string]))
-
-(defn file->seq
-  [filename]
-  (->> (slurp filename)
-       (string/split-lines)))
-
-(defn next-location
-  [loc [right down]]
-  [(+ (first loc) right)
-   (+ (second loc) down)])
-
-(defn locations-s
-  [height [right down]]
-  (take-while #(< (second %) height)
-              (map #(next-location [0 0] [(* right %) (* down %)])
-                   (range))))
-
-(defn tree?
-  [ch]
-  (= ch \#))
-
-(defn count-trees-on-slope
-  [grid [right down]]
-  (let [height          (count grid)
-        valid-locations (locations-s height [right down])]
-    (->> valid-locations
-         (map #(get-in grid [(second %) (mod (first %)
-                                             (count (first grid)))]))
```

Comment:
sloppy syntax and imprecise selection , rewrite as a full (fn)

Reply:
> sloppy syntax and imprecise selection , rewrite as a full (fn)

rewrote in second PR.